Mathlib/Analysis/Calculus/FDeriv/Pow.lean

```diff
@@ -0,0 +1,221 @@
+/-
+Copyright (c) 2025 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.Analysis.Calculus.FDeriv.Mul
+import Mathlib.Analysis.Calculus.FDeriv.Comp
+
+/-!
+# Derivative of `(f x) ^ n`, `n : ℕ`
+
+In this file we prove that `(x ^ n)' = n * x ^ (n - 1)`, where `n` is a natural number.
+Additionally, we prove the case for non-commutative rings,
+where the result is instead `∑ i ∈ Finset.range n, f x ^ (n.pred - i) •> f' <• f x ^ i`.
```

Comment:
This paragraph is strange because the first sentence is about the derivative of `x -> x ^ n`, and the second about `x -> (f x) ^ n`.

Reply:
Also, could you mention in the file-level docstring that primed lemmas refer to the noncommutative version?